Resources/Private/BackendModule/src/Components/ItemsList.tsx

```diff
@@ -5,26 +5,30 @@ import {AssetListItem} from "./index";
 import StateInterface from "../Store/StateInterface";
 import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
 import {faSpinner} from "@fortawesome/free-solid-svg-icons";
-import BackendAssetModuleResultDtoInterface from "../Model/BackendAssetModuleResultDtoInterface";
 import PureComponent from "./PureComponent";
+import FocusKeywordListItem from "./FocusKeywordListItem";
+import {StatefulModuleItem} from "../Model/StatefulModuleItem";
 
 @connect((state: StateInterface) => ({
-    assets: state.app.items,
+    scope: state.app.scope,
+    items: state.app.items,
     loading: state.app.loading,
     started: state.app.started,
     hasError: state.app.hasError,
```

Comment:
Du hast hier mehrere States loading, started, hasError, ... in vielen Komponenten checkst du mehrere Kombinationen davon und die Logik ist quer über viele Komponenten verteilt, soweit ich gesehen hab gibt:

 - hasError:true dann soll nur eine Fehlermeldung angezeigt werden, und der ReturnToOverviewButton
 - (!started && !hasError) dann soll das Formular angezeigt werden, und der StartModuleButton
 - (started && !hasError) dann soll die Liste angezeigt werden
    -  loading:true dann loadingIndicator 
    - (Object.keys(items).length === 0 && !loading) dann emptyListIndicator, und ReturnToOverviewButton (warum nicht zurück zum ConfigurationForm?)
    - Immer renderList (wenn hasItems, dann auch SubmitAndFetchNextButton)
    
Was auffällt, es gibt keine allgemeinen Buttons, also würden die innerhalb des einzelenen Falls viel einfacher zu rendern sein ohne separate condition.

 Obiges Ding erlaubt jede Menge fehlerhafter Zustände, was ist z.B. wenn started:false lnd loading:true?
 
Es gibt ein Pattern um so etwas leicht vorhersehbar zu machen, das heißt StateMachiene